include/oneapi/dpl/pstl/parallel_backend_omp.h

```diff
@@ -0,0 +1,221 @@
+// -*- C++ -*-
+// -*-===----------------------------------------------------------------------===//
+//
+// Copyright (C) Christopher Nelson
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_PARALLEL_BACKEND_OMP_H
+#define _ONEDPL_PARALLEL_BACKEND_OMP_H
+
+#include <algorithm>
+#include <atomic>
+#include <iterator>
+#include <cstddef>
+#include <cstdio>
+#include <memory>
+#include <vector>
+#include <omp.h>
+
+#include "parallel_backend_utils.h"
+#include "unseq_backend_simd.h"
+#include "utils.h"
+
+#if !defined(_OPENMP)
+#    error _OPENMP not defined.
+#endif
+
+// Portability "#pragma" definition
+#ifdef _MSC_VER
+#    define _PSTL_PRAGMA(x) __pragma(x)
+#else
+#    define _PSTL_PRAGMA(x) _Pragma(#    x)
+#endif
+
+namespace oneapi
+{
+namespace dpl
+{
+namespace __omp_backend
+{
+
+//------------------------------------------------------------------------
+// use to cancel execution
+//------------------------------------------------------------------------
+inline void
+__cancel_execution()
```

Comment:
I guess we may implement the function via "#pragma omp cancel"... 
Have you tried to do it?

Reply:
No. How would I know what to cancel? Sometimes we use for, sometimes we use tasks... Should I always use "cancel parallel" ?

Reply:
Actually, I did not use the pragma... Probably we cannot do cancelation for all scenarios.. 
For now I would  suggest writing //TODO: To consider usage of #pragma omp cancel

Reply:
Also, a naked `_PSTL_PRAGMA(omp cancel parallel)` doesn't compile. So, I'm not sure in general how to do this.